tests/test_stroke.py

```diff
@@ -23,6 +23,10 @@ def test_is_back(self):
         self.assertEqual(HANDSTROKE.is_back(), False)
         self.assertEqual(BACKSTROKE.is_back(), True)
 
+    def test_from_index(self):
+        for i in range(-100, 100):
+            self.assertEqual(Stroke.from_index(i).is_hand(), i % 2 == 0)
+
```

Comment:
Currently this test just repeats the `i % 2 == 0` from the implementation. Also I don't think any failing value of `i` would be reported, just a message that `True != False`?

I think it would be clearer to assert some specific cases to help make sure it's the right way around.
```python
def test_from_index__first_row__is_handstroke(self):
    self.assertEqual(Stroke.from_index(0), HANDSROKE)

def test_from_index__second_row__is_backstroke(self):
    self.assertEqual(Stroke.from_index(1), BACKSROKE)

def test_from_index__third_row__is_handstroke(self):
    self.assertEqual(Stroke.from_index(2), HANDSROKE)
```

Reply:
OK this is a good shout.  I think I'll keep the `i % 2 == 0` test, since we don't lose anything by over-testing some code.

Reply:
Fixed, I think